snmp/datadog_checks/snmp/utils.py

```diff
@@ -92,3 +95,69 @@ def oid_pattern_specificity(pattern):
         len(parts),  # Shorter OIDs are less specific than longer OIDs, regardless of their contents.
         parts,  # For same-length OIDs, compare their contents (integer parts).
     )
+
+
+class OIDPrinter(object):
+    """Utility class to display OIDs efficiently.
+
+    This is only meant for debugging, as it makes some assumptions on the data
+    managed, and can be more expensive to use than regular display.
+    """
+
+    def __init__(self, oids, with_value):
+        self.oids = oids
+        self.with_value = with_value
+
+    def oid_str(self, oid):
+        try:
+            return oid[0].getOid().prettyPrint()
+        except SmiError:
+            arg = oid._ObjectType__args[0]._ObjectIdentity__args[-1]
```

Comment:
Wow, this is gnarly, great example of why we need this helper class!

Reply:
The reason _why_ Thomas had to do this is even more gnarly…

It turns out that PySNMP `ObjectType` objects implement `__getitem__()` to expose a 2-tuple interface, where essentially `oid[0]` is the OID itself (an `ObjectIdentity`) and `oid[1]` is the value.

_But_… if the OID doesn't have a value, then accessing the OID itself (as here) will fail with `SmiError: OID is not initialized`.

I remember I came across this back in #5531 — I'll try to submit parts of that refactor as incremental PRs next week — and worked around it with the `OID` model class. :-)

Anyway, all PySNMP madness aside — _maybe_ we can add a small comment to explain this workaround (as otherwise it looks a bit jarring)?

```suggestion
            # PySNMP screams when we try to access the name of an OID
            # that has no value yet. Fine, let's work around this arbitrary limitation...
            arg = oid._ObjectType__args[0]._ObjectIdentity__args[-1]
```